pkg/operator/observe_config.go

```diff
@@ -280,28 +280,34 @@ func observeRestrictedCIDRs(listers Listers, currentConfig map[string]interface{
 
 // observeInternalRegistryHostname reads the internal registry hostname from the cluster configuration as provided by
 // the registry operator.
-func observeInternalRegistryHostname(listers Listers, currentConfig map[string]interface{}) (observedConfig map[string]interface{}, errs []error) {
-	observedConfig = map[string]interface{}{}
+func observeInternalRegistryHostname(listers Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+
 	internalRegistryHostnamePath := []string{"imagePolicyConfig", "internalRegistryHostname"}
-	currentInternalRegistryHostname, _, _ := unstructured.NestedStringSlice(currentConfig, internalRegistryHostnamePath...)
+	if currentInternalRegistryHostname, _, _ := unstructured.NestedString(existingConfig, internalRegistryHostnamePath...); len(currentInternalRegistryHostname) > 0 {
+		unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
```

Comment:
@sanchezl this structure makes it a little easier to follow the logic of "return the previous value"

Reply:
:+1: